src/coreclr/vm/syncblk.inl

```diff
@@ -650,7 +654,11 @@ FORCEINLINE AwareLock::EnterHelperResult ObjHeader::EnterObjMonitorHelper(Thread
         return AwareLock::EnterHelperResult_UseSlowPath;
     }
 
+#if defined(TARGET_WINDOWS) && defined(TARGET_ARM64)
+    if (FastInterlockedCompareExchangeAcquire((LONG*)&m_SyncBlockValue, newValue, oldValue) == oldValue)
```

Comment:
Didn't we just remove some or all of the "FastInterlockedXXXX" versions?

Having "Fast" variants always feels a bit confusing. Are they unfit for some uses? Where would want to be using the slow versions?

Reply:
Yes, we wanted to just patch the hot paths (which is where we should always use fast versions) instead of all the paths and hence I had to reintroduce them at just few call-sites.